tests/unit/v2/testClient.js

```diff
@@ -0,0 +1,143 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const { generateFakeEvents } = require('../../utils/v2Data');
+
+const UtapiClient = require('../../../libV2/client');
+
+const events = generateFakeEvents(1, 50, 50);
+const sortEvent = (a, b) => (a.uuid > b.uuid ? 1 : -1);
+
+// eslint-disable-next-line func-names
+describe('Test UtapiClient', function () {
+    this.timeout(10000);
+    let client;
+    let sandbox;
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        client = new UtapiClient({
+            drainDelay: 10,
+        });
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+        client = null;
+    });
+
+    it('should push a metric', next => {
+        sandbox.stub(UtapiClient.prototype, '_pushToUtapi').resolves();
+        client.pushMetric(events[0], err => {
+            assert.strictEqual(err, null);
```

Comment:
you may use `assert.ifError(err)`, it would also accept `undefined` and might have a better error reporting